### PR TITLE
Added a Scikit-learn like `train_test_split` method

### LINKED ISSUE #8: Make a general function for all splitting strategies.
Similar to the [sklearn.train_test_split](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html) method, implement another function `general_split` or `train_test_split` that does any kind of splitting.

```python
def general_split(
    mols: Union[datamol.Mol, str], 
    test_size: Union[float, int], 
    splitting_method: Literal["random", "scaffold", "kmeans"], 
    random_state: int = 42, 
    n_jobs:int=0, 
    *args, 
    **kwargs)

    print("Do some magic")
    return train_idx, test_idx
```

### PR DESCRIPTION
**Pull request by cwognum**
## Changelogs

- Closes #8 by adding a general `train_test_split` method

---

_Checklist:_

- [X] _Was this PR discussed in an issue? It is recommended to first discuss a new feature into a GitHub issue before opening a PR._
- [X] _Add tests to cover the fixed bug(s) or the newly introduced feature(s) (if appropriate)._
- [x] _Update the API documentation if a new function is added, or an existing one is deleted._
- [X] _Write concise and explanatory changelogs above._
- [X] _If possible, assign one of the following labels to the PR: `feature`, `fix` or `test` (or ask a maintainer to do it for you)._

---

You can now do: 

```python
from splito import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y, smiles=smiles, method="scaffold")
```

### CODE REVIEW

File: splito/_split.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    SCAFFOLD = ScaffoldSplit
+    STRATIFIED_DISTRIBUTION = StratifiedDistributionSplit
+    MOLECULAR_WEIGHT = MolecularWeightSplit
+    MIN_MAX_DIVERSITY_SPLIT = MolecularMinMaxSplit
+
+
+def train_test_split(
+    X: np.ndarray,
+    y: np.ndarray,
+    molecules: Optional[Sequence[Union[str, dm.Mol]]] = None,
```
Comment (DomInvivo) [inline splito/_split.py:32]: I understand here the goal is to mimic scikit learn, but I don't really understand what is X in that context? It seems to only support arrays, i.e. if the molecules are represented as vectors.

What if someone wants to use PyG objects? Or what if someone just wants to split a set of smiles to create fixed CSV's? Then this function doesn't work and providing `X` is ambiguous.

It would be ideal if you can just provide the `train_indices` and `test_indices` and let the person split afterwards.

Of course, you could say, "If the user wants indices, they can provide `X` as indices", but it's not straightforward and confusing.

File: splito/_split.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    method: Union[str, SimpleSplittingMethod] = "random",
```
Comment (DomInvivo) [inline splito/_split.py:33]: No support for custom callables for the splitting method?
Not sure if that defeats the purpose of `Splito`

File: splito/_split.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    X = np.array(X)
+    y = np.array(y)
+
+    method = SimpleSplittingMethod[method.upper()] if isinstance(method, str) else method
+
+    splitter_kwargs = {"test_size": test_size, "random_state": seed}
+    if method in [
+        SimpleSplittingMethod.MOLECULAR_WEIGHT,
+        SimpleSplittingMethod.MIN_MAX_DIVERSITY_SPLIT,
+        SimpleSplittingMethod.SCAFFOLD,
```
Comment (DomInvivo) [inline splito/_split.py:89]: Then you add a new method, and forget to list it here, and everything breaks.

Is it possible to check if the signature of the function requires `smiles` or `mol` or something?

File: splito/_split.py (hunk 1 above)
Comment (DomInvivo) [reply splito/_split.py:32]: An alternative is to have another function `train_test_split_indices` that is called by this one.

Comment (cwognum) [reply splito/_split.py:32]: I like your second suggestion!

File: splito/_split.py (hunk 2 above)
Comment (cwognum) [reply splito/_split.py:33]: I think that defeats the purpose of Splito. If users request it, I would be happy to consider ways of adding this, but for now I think it's best to minimize complexity.

File: splito/_split.py (hunk 3 above)
Comment (cwognum) [reply splito/_split.py:89]: That's what test cases are for! 😉 

In all seriousness, I agree that this is not the most sophisticated solution. In the short-term, however, I don't expect a lot of contributors and would prefer to keep simple and only introduce complexity once we have a better idea of the use cases we want to support.
